### PR TITLE
Add mtu configuration to sriov_pfs and bonds

### PR DESCRIPTION
**Pull request by wmousa**
Allow seting mtu for sriov_pfs and bonds, but please note that:
  - Configure mtu for sriov_pf will be ignored if sriov_pf is a bond slave
  - Configure mtu for bond interface will configure it to its slaves
  - Configure mtu for sriov_pf will set it to the brdige if sriov_pf is ovs port

### CODE REVIEW

**Comment by wmousa** [comment]
@moshe010 @adrianchiris

File: switchdev_utils/config-service/sample_config.yaml [hunk 1]
```diff
@@ -8,24 +8,25 @@ pfs:
...
+    mtu: 9000
 linuxBonds:
   - name: bond0
     slaves: [ens2f0, ens2f1]
     bondingOptions: "mode=802.3ad miimon=100 lacp_rate=fast xmit_hash_policy=layer2+3"
-  - name: bond1
-    slaves: [ens1f0, ens1f1]
-    bondingOptions: "mode=802.3ad miimon=100 lacp_rate=fast xmit_hash_policy=layer2+3"
+    mtu: 9000
+
```
Comment (adrianchiris) [inline switchdev_utils/config-service/sample_config.yaml:27]: Any reason to change the configuration from bond to pf ?

File: switchdev_utils/config-service/switchdev_network_config [hunk 2]
```diff
@@ -566,15 +570,23 @@ def validate_config(config):
             logger.error(err)
             raise ValueError(err)
         try:
-            if int(sriov_pf.get("numOfVfs")):
-                continue
+            int(sriov_pf.get("numOfVfs"))
         except ValueError as value_err:
-            logger.error("numOfVfs for %s must be number" %
-                         sriov_pf.get("name"))
+            err = "numOfVfs for %s must be number" % sriov_pf.get("name")
```
Comment (adrianchiris) [inline switchdev_utils/config-service/switchdev_network_config]: why do you need this assignment ? and not just logger.error("numOfVfs....")

File: switchdev_utils/config-service/switchdev_network_config [hunk 3]
```diff
@@ -566,15 +570,23 @@ def validate_config(config):
...
+            logger.error(err)
             raise value_err
         except TypeError as type_err:
-            logger.error("numOfVfs for %s must be provided" %
-                         sriov_pf.get("name"))
+            err = "numOfVfs for %s must be provided" % sriov_pf.get("name")
```
Comment (adrianchiris) [inline switchdev_utils/config-service/switchdev_network_config]: same comment as above

File: switchdev_utils/config-service/switchdev_network_config [hunk 4]
```diff
@@ -566,15 +570,23 @@ def validate_config(config):
...
+            logger.error(err)
+            raise type_err
+        if sriov_pf.get("mtu"):
+            try:
+                int(sriov_pf.get("mtu"))
+            except ValueError as value_err:
+                err = "mtu %s must be number" % sriov_pf.get("mtu")
+                logger.error(err)
```
Comment (adrianchiris) [inline switchdev_utils/config-service/switchdev_network_config]: same as above

File: switchdev_utils/config-service/switchdev_network_config [hunk 5]
```diff
@@ -589,12 +601,19 @@ def validate_config(config):
...
         else:
             err = "slaves of bond %s must be list of sriov_pfs" % \
                   bond.get("name")
             logger.error(err)
             raise ValueError(err)
+        if bond.get("mtu"):
+            try:
+                int(bond.get("mtu"))
+            except ValueError as value_err:
+                err = "mtu %s must be number" % bond.get("mtu")
```
Comment (adrianchiris) [inline switchdev_utils/config-service/switchdev_network_config]: same as above

File: switchdev_utils/config-service/README.md [hunk 6]
```diff
@@ -33,11 +33,13 @@ List of SRIOV PF objects which need to be configured, and has the following attr
     - numOfVfs: Number of vfs to be configured on the sriov pf.  
     - switchdev: Boolean, Enable switchdev mode.  
     - vfTrust: Boolean, Configure vf trust.  
+    - mtu: Number, configure mtu for sriov_pf (it will be ignored if sriov_pf is a bond slave)
 - linuxBonds:  
 list of linux bond objects which need to be created and configured  
     - name: The name of the bond interface  
     - slaves: List of bond slaves interfaces (a subset of sriov pfs)  
     - bondingOptions: Bonding options of the bond interface  
+    - mtu: Number, configure mtu for bond interface and its slaves
```
Comment (adrianchiris) [inline switchdev_utils/config-service/README.md:42]: can you add a Note at the end of this section explaining the mtu behaviour when set on bond / pfs as describe in commit message ?

**Review by adrianchiris — CHANGES_REQUESTED** [review]

File: switchdev_utils/config-service/sample_config.yaml (hunk 1 above)
Comment (wmousa) [reply switchdev_utils/config-service/sample_config.yaml:27]: @adrianchiris 
I wanted to cover all cases that the script is doing and especially for mtu:
- The sriov_pf could be a bond slave so only setting mtu to the bond needed, no need to set it to the slaves
- The sriov_pf could be an ovs port, so setting mtu only for it, no need to set it to the bridge
- The sriov_pf could be single interface, so setting mtu to it directly

File: switchdev_utils/config-service/switchdev_network_config (hunk 2 above)
Comment (wmousa) [reply switchdev_utils/config-service/switchdev_network_config]: done

File: switchdev_utils/config-service/switchdev_network_config (hunk 3 above)
Comment (wmousa) [reply switchdev_utils/config-service/switchdev_network_config]: done

File: switchdev_utils/config-service/switchdev_network_config (hunk 4 above)
Comment (wmousa) [reply switchdev_utils/config-service/switchdev_network_config]: done

File: switchdev_utils/config-service/README.md (hunk 6 above)
Comment (wmousa) [reply switchdev_utils/config-service/README.md:42]: done

File: switchdev_utils/config-service/switchdev_network_config (hunk 5 above)
Comment (wmousa) [reply switchdev_utils/config-service/switchdev_network_config]: done

File: switchdev_utils/config-service/switchdev_network_config [hunk 7]
```diff
@@ -589,12 +600,18 @@ def validate_config(config):
             raise ValueError(err)
         if (bond.get("slaves") and
                 (set(bond.get("slaves")).issubset(set(sriov_pfs_list)))):
-            continue
+            pass
```
Comment (moshe010) [inline switchdev_utils/config-service/switchdev_network_config]: you can remove the pass by doing also put a comment explaining this if statement 
if (not (bond.get("slaves") and
                (set(bond.get("slaves")).issubset(set(sriov_pfs_list))))):
            err = "slaves of bond %s must be list of sriov_pfs" % \
                  bond.get("name")	                  bond.get("name")
            logger.error(err)	            logger.error(err)
            raise ValueError(err)	            raise ValueError(err)

Comment (wmousa) [reply switchdev_utils/config-service/switchdev_network_config]: done

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**